### ISSUE TITLE
docker-compose up: no working instance

### ISSUE THREAD

**Issue by mrballer** [PrestaShop/PrestaShop#11231]
**Describe the bug**

Error Message: prestashop-git | \n* Waiting for confirmation of MySQL service startup

**To Reproduce**
1. git clone https://github.com/PrestaShop/PrestaShop.git
2. cd PrestaShop
3. docker-compose up

**Comment by firstred** [comment]
This could be related to the fact that the `docker-compose.yml` does not have a fixed port. 

You can try mapping the port `"3306"` => `"33068:3306"` for example.

Also, what does your environment look like? Is it recent enough? 

With 
```shell
$ docker -v
Docker version 17.12.1-ce, build 7390fc6
$ docker-compose -v
docker-compose version 1.17.1, build unknown
``` 
I had no problems.

**Comment by Quetzacoalt91** [comment]
Binding the port is important when you want to reach the sql instance from your host, this is unlikely causing the issue here.
Have you modified the docker-compose.yml before starting it? A network is supposed to be created when you start the stack in which the containers can see each other. I've tested and the mysql container is reachable.

For instance here:
```
$ docker network inspect prestashop_default_1

[...]
        "Containers": {
            "5e9e5a4497f6f90603b8ba1afdb0da547abd29534e544e2dbc3019cbf16d37c3": {
                "Name": "prestashop_mysql_1",
                "EndpointID": "272c1bf050627289b064ace59aae46545bf5a5d09fade75f81c9d5f5b624a01e",
                "MacAddress": "02:42:ac:12:00:02",
                "IPv4Address": "172.18.0.2/16",
                "IPv6Address": ""
            },
            "62753be17eb5de5bc133a29059a94799d6413b9e45b2ceb92c9e7aa4818a8010": {
                "Name": "prestashop-git",
                "EndpointID": "2b52cfb04ed68b61aab803969929aeed91d643b1f43bed349210e6be29b3cb66",
                "MacAddress": "02:42:ac:12:00:03",
                "IPv4Address": "172.18.0.3/16",
                "IPv6Address": ""
            }
        },
```

```
$ docker exec -ti prestashop-git ping mysql
PING mysql (172.18.0.2): 56 data bytes
64 bytes from 172.18.0.2: icmp_seq=0 ttl=64 time=0.088 ms
64 bytes from 172.18.0.2: icmp_seq=1 ttl=64 time=0.088 ms
64 bytes from 172.18.0.2: icmp_seq=2 ttl=64 time=0.097 ms
64 bytes from 172.18.0.2: icmp_seq=3 ttl=64 time=0.078 ms
--- mysql ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max/stddev = 0.078/0.088/0.097/0.000 ms
```

**Comment by khouloudbelguith** [comment]
Hi @nfmaroc,

I think that the issue related to the MySQL tag is fixed by https://github.com/PrestaShop/docker/pull/129
Please be aware that the project should be used to report issues related to the Prestashop core.
You need to create an issue on this project: https://github.com/PrestaShop/docker/issues
Thanks for your report.

**Comment by khouloudbelguith** [comment]
Hi @nfmaroc,

Since we had no news from you for more than 30 days, I'll close this ticket. Feel free to open another one if you can give specific details.
Thanks!

**Comment by BachmannKarlHendrik** [comment]
I ran into the same error today, using this docker-compose:
https://github.com/PrestaShop/PrestaShop/commit/16b209d5694445ece3a316d7c802a30e05d78c79#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3
(this is currently the latest docker-compose.yml file in 1.7.8.x branch)

As it seems to me, mysql server does not honor the root password that is set in the docker-compose file.

First, I tried:
`docker exec -it prestashop-prestashop-git-1 mysql -h mysql -u prestashop -p`
`ERROR 1130 (HY000): Host '172.18.0.3' is not allowed to connect to this MySQL server`

So I tried to log in into mysql server, but it was not possible with the `prestashop` password.


`SET PASSWORD FOR 'root'@'localhost' = PASSWORD('prestashop');
GRANT ALL PRIVILEGES ON *.* TO 'root'@'172.19.0.3' IDENTIFIED BY 'prestashop' WITH GRANT OPTION;
FLUSH PRIVILEGES;`
Fixed the issue for me.
